### DISCUSSION TITLE
Attitude towards Serverless Integration (willing to contribute)

### DISCUSSION THREAD

**Discussion by darioackermann** [dani-garcia/vaultwarden#3897] [General]
Hey! I'm aiming to bring AWS Serverless Support to vaultwarden. I've done a small POC today and so far, everything seems working well. 
I will resolve small teething issues in the coming days.

I was wondering if some small changes to provide AWS Lambda support, for example, would be accepted? The only change, apart from a few usage declarations, is
```rs
if is_running_on_lambda() {
    // Launch on AWS Lambda
    launch_rocket_on_lambda(instance).await?;
} else {
    // Launch local server
    let _ = instance.launch().await?;
}
```

I'd prefer this solution over having to fork vaultwarden and keeping it in sync the entire time, also making it easier for anyone else who is interested on running it on serverless infrastructure.

Your feedback is gladly appreciated! :)

**Comment by BlackDex** [comment]
What are all the other changes needed to be done? Is there are fork which we can check?
Because this of course will not work out-of-the-box.

Also, it would be an extra item to maintain for us, and not easy to test and debug.

**Reply by darioackermann** [reply]
Thanks! I started a fork and opened a pull request in the fork so you have an easy way to see the changes without spamming this repo with a PR: https://github.com/darioackermann/vaultwarden-serverless/pull/1

There are a few things left to check but I don't expect many more changes to the rust source...

**Comment by marksteele** [comment]
A write up on how to set this up: https://www.control-alt-del.org/posts/privacy-and-the-browser/
